cmd/account/mgmt/account-assign.go

```diff
@@ -0,0 +1,188 @@
+	OSDStaging1OuID   = "ou-0wd6-z6tzkjek"
+)
+
+type accountAssignOptions struct {
+	awsClient    awsprovider.Client
+	username     string
+	payerAccount string
+	output       string
+
+	flags      *genericclioptions.ConfigFlags
+	printFlags *printer.PrintFlags
+	genericclioptions.IOStreams
+}
+
+func newAccountAssignOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *accountAssignOptions {
+	return &accountAssignOptions{
+		flags:      flags,
+		printFlags: printer.NewPrintFlags(),
+		IOStreams:  streams,
+	}
+}
+
+// assignCmd assigns an aws account to user under osd-staging-2 by default unless osd-staging-1 is specified
+func newCmdAccountAssign(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newAccountAssignOptions(streams, flags)
+	accountAssignCmd := &cobra.Command{
+		Use:               "assign",
+		Short:             "Assign account to user",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	ops.printFlags.AddFlags(accountAssignCmd)
+	accountAssignCmd.Flags().StringVarP(&ops.payerAccount, "payer-account", "p", "", "Payer account type")
+	accountAssignCmd.Flags().StringVarP(&ops.username, "username", "u", "", "LDAP username")
+
+	return accountAssignCmd
+}
+
+func (o *accountAssignOptions) complete(cmd *cobra.Command, _ []string) error {
+	if o.username == "" {
+		return cmdutil.UsageErrorf(cmd, "LDAP username was not provided")
+	}
+	if o.payerAccount == "" {
+		return cmdutil.UsageErrorf(cmd, "Payer account was not provided")
+	}
+
+	return nil
+}
+
+func (o *accountAssignOptions) run() error {
+
+	var (
+		accountAssignID string
+		destinationOU   string
+		rootID          string
+	)
+
+if o.payerAccount == "osd-staging-1" {
+    rootID = OSDStaging1RootID
+    destinationOU = OSDStaging1OuID
+} else if o.payerAccount == "osd-staging-2" {
+ 	rootID = OSDStaging2RootID
+	destinationOU = OSDStaging2OuID
+} else {
+    return fmt.Errorf("Invalid payer account provided")
+}
+	//Instantiate aws client
+	awsClient, err := awsprovider.NewAwsClient(o.payerAccount, "us-east-1", "")
+	if err != nil {
+		return err
+	}
+
+	o.awsClient = awsClient
+	accountAssignID, err = o.findUntaggedAccount()
+	if err != nil {
+		return err
+	}
+
+	err = o.tagAccount(accountAssignID)
+	if err != nil {
+		return err
+	}
+
+	err = o.moveAccount(accountAssignID, destinationOU, rootID)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(o.IOStreams.Out, accountAssignID)
+
+	return nil
+}
+
+var ErrNoUntaggedAccounts = fmt.Errorf("no untagged accounts available")
+var ErrNoAccountsInRoot = fmt.Errorf("no accounts available in root")
+
+func (o *accountAssignOptions) findUntaggedAccount() (string, error) {
+	//List accounts that are not in any OU
+	input := &organizations.ListAccountsInput{}
+	accounts, err := o.awsClient.ListAccounts(input)
```

Comment:
I think we need to use the `ListAccountsForParent` function, passing in the root ou here.  Otherwise we get _all_ accounts, whether they're in a different OU or not, which will throw an error when attempting to "Move Account" later because AWS will say the account "doesn't exist" but it doesn't exist in the root OU, it's in another OU.